dev/Common/Microsoft.RoApi.h

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+#ifndef __MICROSOFT_ROAPI_H
+#define __MICROSOFT_ROAPI_H
+
+#include <roapi.h>
+#include <wrl.h>
+#include <wil/resource.h>
+
+namespace Microsoft::RoApi
+{
+    // WinAppSDK Components that activate a class registered via windows.activatableClass.proxyStub
+    // in manifest should call this function before due to a RoActivateInstance caching bug.
+    inline void ClearRoActivateInstanceCache()
+    {
+        // Activating a nonexistent class clears the cache enables the proxyStub to be found for elevated scenarios
+        auto activatableClass{ wil::make_unique_string_nothrow<wil::unique_hstring>(L"FakeClass.ClassName") };
+        winrt::com_ptr<::IInspectable> ptr{};
+
+        // Function is expected fail and if anything else happens we should throw.
+        winrt::hresult hr{ RoActivateInstance(activatableClass.get(), ptr.put()) };
+        THROW_HR_IF(hr, hr != REGDB_E_CLASSNOTREG);
```

Comment:
You probably want to assert that hr didn't succeed first. Unlikely, but someone could actually define a class with the name you put in above and the activation passes.

Reply:
Also if I understand correct, you cannot THROW S_OK. The macro would complain.

Reply:
Yup, just added another error line.